osidb/models.py

```diff
@@ -1808,6 +1845,76 @@ class Meta:
             "created_dt",
         )
 
+    def bzsync(self, *args, bz_api_key=None, **kwargs):
+        """
+        Bugzilla sync of the FlawComment instance and of the related Flaw instance.
+        """
+
+        self.save()
+
+        # Comments need to be synced through flaw
+        # If external_system_id is blank, BugzillaSaver posts the new comment
+        # and FlawCollector loads the new comment and updates this FlawComment
+        # instance to match bugzilla.
+        self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
+
+    def _validate_single_pending_comment(self):
+        """
+        Check that this is at most the first pending comment for the flaw.
+        This comment must not be a second pending comment for the flaw. If it
+        is, it means that API calls are not ATOMIC_REQUESTS anymore, or that
+        the temporary pending comments are not cleaned up properly, or that
+        something makes concurrent requests.
+        """
+        tmp_cmnts = FlawComment.objects.pending().filter(flaw=self.flaw)
+        if tmp_cmnts.count() > 1:
+            raise ValidationError(
+                f"FlawComment ({self.uuid}) for {self.flaw} is not the first "
+                "pending new comment."
+            )
+
+    def _validate_comment_order(self):
+        """
+        Check that this comment's order (comment number) is directly following
+        the last existing comment of the given flaw. This checks that the
+        client has a sufficiently up-to-date idea of the flaw when submitting
+        the comment.
+
+        Usually, a flaw has a comment number 0 as its description. However,
+        if the description is missing, the first non-description comment must
+        start with number 1.
+        """
+        if self.flaw.comments.count() == 0:
+            proper_following_order = 1
+        else:
+            proper_following_order = (
+                1 + self.flaw.comments.aggregate(Max("order"))["order__max"]
+            )
+        if self.external_system_id == "" and proper_following_order != self.order:
+            raise ValidationError(
+                f"FlawComment ({self.uuid}) for {self.flaw} is not of the correct "
+                f"order. FlawComment.order ({self.order}) but flaw has "
+                f"({self.flaw.comments.count()}) comments (starting with 0). "
+                "Either OSIDB's DB is outdated or the client has an outdated "
```

Comment:
Can the DB be more outdated than the client when the client gets data from the DB and not vise versa?

Reply:
I imagine three scenarios are possible:

- Client has information from OSIDB that matches OSIDB database, but OSIDB itself has an outdated information about the flaw (bugzilla was updated from elsewhere recently).

- Client has information from OSIDB, but after it got that information, OSIDB fetched an update that the client doesn't have yet.

- The two above, combined.

Does that seem right?

Reply:
I probably did not understand the alert. I thought it is just between the client and OSIDB. If not so then never mind and you can leave it as it is.

Reply:
I'm enhancing the docstring of this validation based on this discussion.

Reply:
Resolved in https://github.com/RedHatProductSecurity/osidb/pull/209/commits/f9ebfa5badd141532da083c3d6506b796e4868b4